src/gui/accountsettings.cpp

```diff
@@ -334,40 +334,43 @@ void AccountSettings::slotEditCurrentIgnoredFiles()
     openIgnoredFilesDialog(f->path());
 }
 
-void AccountSettings::slotOpenMkFolderDialog()
+void AccountSettings::slotOpenMakeFolderDialog()
 {
-    const auto selected = _ui->_folderList->selectionModel()->currentIndex();
+    const QString fileName = [this] () {
+        const auto selected = _ui->_folderList->selectionModel()->currentIndex();
 
-    if (!selected.isValid()) {
-        qCWarning(lcAccountSettings) << "Selection model current folder index is not valid.";
-        return;
-    }
+        if (!selected.isValid()) {
+            qCWarning(lcAccountSettings) << "Selection model current folder index is not valid.";
+            return QString();
+        }
 
-    const auto classification = _model->classify(selected);
+        const auto classification = _model->classify(selected);
 
-    if (classification != FolderStatusModel::SubFolder && classification != FolderStatusModel::RootFolder) {
-        return;
-    }
+        if (classification != FolderStatusModel::SubFolder && classification != FolderStatusModel::RootFolder) {
+            return QString();
+        }
 
-    QString fileName;
+        QString result;
+        if (classification == FolderStatusModel::RootFolder) {
+            const auto alias = _model->data(selected, FolderStatusDelegate::FolderAliasRole).toString();
+            const auto folderMan = FolderMan::instance();
+            if (const QPointer<Folder> folder = folderMan->folder(alias)) {
+                result = folder->path();
+            }
+        } else {
+            result = _model->data(selected, FolderStatusDelegate::FolderPathRole).toString();
+        }
 
-    if (classification == FolderStatusModel::RootFolder) {
-        const auto alias = _model->data(selected, FolderStatusDelegate::FolderAliasRole).toString();
-        const auto folderMan = FolderMan::instance();
-        if (const QPointer<Folder> folder = folderMan->folder(alias)) {
-            fileName = folder->path();
+        if (!result.isEmpty()) {
+            if (result.endsWith('/')) {
+                result.chop(1);
+            }
         }
-    } else {
-        fileName = _model->data(selected, FolderStatusDelegate::FolderPathRole).toString();
-    }
+        return result;
+    }();
 
     if (!fileName.isEmpty()) {
-        if (fileName.endsWith('/')) {
-            fileName.chop(1);
-        }
-        FolderCreationDialog folderCreationDialog;
-        folderCreationDialog.setDestination(fileName);
-        folderCreationDialog.exec();
+        FolderCreationDialog(fileName).exec();
```

Comment:
Note it is still on the stack, without parent and using exec. It should really go the other way around and be on the heap with a parent and using show + delete on close attribute.

Reply:
@er-vin Changed